core/domain/subtopic_page_services.py

```diff
@@ -86,7 +101,44 @@ def get_subtopic_page_from_model(subtopic_page_model):
     )
 
 
-def get_subtopic_page_by_id(topic_id, subtopic_id, strict=True):
+@overload
+def get_subtopic_page_by_id(
+    topic_id: str, subtopic_id: int
+) -> subtopic_page_domain.SubtopicPage: ...
+
+
+@overload
+def get_subtopic_page_by_id(
+    topic_id: str,
+    subtopic_id: int,
+    *,
+    strict: Literal[True]
+) -> subtopic_page_domain.SubtopicPage: ...
+
+
+@overload
+def get_subtopic_page_by_id(
+    topic_id: str,
+    subtopic_id: int,
+    *,
+    strict: Literal[False]
+) -> Optional[subtopic_page_domain.SubtopicPage]: ...
+
+
+@overload
+def get_subtopic_page_by_id(
+    topic_id: str,
+    subtopic_id: int,
+    *,
+    strict: bool = ...
+) -> Optional[subtopic_page_domain.SubtopicPage]: ...
```

Comment:
Sorry, I'm a bit confused why is this needed.

Reply:
https://github.com/oppia/oppia/blob/6f447b0bbf68d0bcdfe75eb211ed2f28bda8bef1/core/domain/subtopic_page_services.py#L154 
In this place, we are passing the `strict` variable ( which is of bool type ). So, that's why i added this overload!!

Reply:
I have added this overload just because this method is used in another method, and there we are providing strict varaible which is of bool type like: 
`get_subtopic_page_by_id(aaa, bbb, strict=strict)`.

In `get_subtopic_page_by_id(aaa, bbb, strict=strict)` we are not using either strict=True or strict=False, which causes MyPy to throw error that no overload is available for int, int, bool.